modules/promotions/admin-menu-items/go-pro-promotion-item.php

```diff
@@ -21,7 +26,8 @@ public function get_parent_slug() {
 	}
 
 	public function get_label() {
-		return esc_html__( 'Upgrade', 'elementor' );
+		$upgrade_text = __( 'Upgrade', 'elementor' );
+		return apply_filters( 'elementor/admin_menu/promotion/upgrade_text', ['upgrade_text' => $upgrade_text ] )['upgrade_text'] ?? $upgrade_text;
```

Comment:
```suggestion
		return apply_filters( 'elementor/admin_menu/promotion/upgrade_text', $upgrade_text ) ?? $upgrade_text;
```
I don't see why this has to be an array if you are filtering just one string here.

Reply:
As mentioned before Uniformity of the Promotion filters abstracted. As things stand I'm not yet happy with the suggested splitting of the filter in this file. There's no real benefit and it makes usage of of promotion filters harder to understand. Creating unique arguments data types for each Promotion filter creates the same problem leading to the needs for individually documentation in what could be really cleaner.

Abstractions like these are ideally meant to hide the complexity and inconsistencies in the underlying implementation which in this case has naming inconsistency and data types.

Reply:
The Admin menu item is different than the other Promotions.

The other Promotions have one config array that you can filter all the items with one filter.

The Admin menu item has a separate Label and URL, with **two different filter names**, and is just a simple string that needs to be returned. You don't have to make this an array if we just need to filter a simple string and assign the string to some key.

Reply:
I don't agree here. From the filter abstraction end this is one and the same thing. It's what abstractions are designed to do - hide underlying complexity. I've reduced this to a string in the function as you asked but the filter and the design of the filter abstraction is core to this PR and not something I'm willing to mutate into endless complexity for devs who implement this filter in the end.